db/view/view_update_generator.cc

```diff
@@ -24,6 +25,64 @@ static inline void inject_failure(std::string_view operation) {
 
 namespace db::view {
 
+class view_update_generator::progress_tracker final : public sstables::read_monitor_generator {
+    class read_monitor final : public  sstables::read_monitor {
+        sstables::shared_sstable _sst;
+        uint64_t& _pending_work;
+    public:
+        virtual void on_read_started(const sstables::reader_position_tracker& tracker) override {}
+
+        virtual void on_read_completed() override {
+            if (auto sst = std::exchange(_sst, {})) {
+                _pending_work -= sst->data_size();
+            }
+        }
```

Comment:
I'm not familiar with this read_monitor thing, so I have to ask: Do we know that on_read_completed() can only be called once? Or maybe we know this just because we know the only thing we ever do with the staging sstable is to read it once, to completion?
Is it possible that one day (or today?) we'll want to split the staging sstable read to multiple reads, and this assumption becomes wrong?